include/tunnler/room_member.h

```diff
@@ -108,31 +113,45 @@ class RoomMember final {
      * Attempts to join a room at the specified address and port, using the specified nickname.
      * This may fail if the username is already taken.
      */
-    void Join(const std::string& nickname, const std::string& server = "127.0.0.1", const uint16_t serverPort = DefaultRoomPort, const uint16_t clientPort = 0);
+    void Join(const std::string& nickname, const std::string& server = "127.0.0.1",
+              const uint16_t serverPort = DefaultRoomPort, const uint16_t clientPort = 0);
 
     /**
      * Leaves the current room.
      */
     void Leave();
-    
+
 private:
-    State state;    // Current state of the RoomMember
-    MemberInformation member_information;
-    RoomInformation room_information;
+    std::atomic<State> state; ///< Current state of the RoomMember.
+    MemberList member_information; ///< Information about the clients connected to the same room as us.
+    RoomInformation room_information; ///< Information about the room we're connected to.
+
+    RakNet::RakPeerInterface* peer; ///< RakNet network interface.
 
-    RakNet::RakPeerInterface* peer;   // RakNet network interface
+    std::string nickname; ///< The nickname of this member.
```

Comment:
Not sure. I feel that the MemberInformation / MemberList stuff is a user-facing (API) based structure.

Consider network activity: If we count network activity (sent WifiPackets / s per RoomMember) to show in the frontend* we'd suddenly have useless fields stored in the RoomMember.

Also it adds a layer of indirection which might make the code slightly more complicated for no reason.

*(which I'm intending so network spammers could be found more easily)